conans/model/build_info.py

```diff
@@ -251,8 +384,9 @@ def deps(self):
     def __getitem__(self, item):
         return self._dependencies[item]
 
-    def update(self, dep_cpp_info, pkg_name):
-        assert isinstance(dep_cpp_info, CppInfo)
+    def update(self, cpp_info, pkg_name):
+        assert isinstance(cpp_info, CppInfo)
+        dep_cpp_info = DepCppInfo(cpp_info)
```

Comment:
This is still running thousands of times for a medium size graph. I would try to define the DepCppInfo() object just once for each conanfile. It can be defined before calling this ``update()`` method, maybe in the ``_propagate_info()`` one, as a conanfile private object (initially None)